The library is difficult to use with a strict TypeScript configuration

I have the following TypeScript code in my own project:

```typescript
// File: src/main.ts

import { CometD } from 'cometd';
import { adapt } from 'cometd-nodejs-client';
adapt();

const client = new CometD();
```

And the following TypeScript configuration (notably the "strict" and "noImplicitAny" options):

```json
// File: tsconfig.json
{
    "include": [
        "src/**/*"
    ],
    "compilerOptions": {
        "target": "es2017",
        "module": "commonjs",
        "baseUrl": "./src",
        "noEmit": true,
        "strict": true,
        "noImplicitAny": true,
        "esModuleInterop": true
    }
}
```

Given the above two things, attempting to call the `adapt()` function from this library results in a build error.

```sh-session
> tsc

src/main.ts:4:23 - error TS7016: Could not find a declaration file for module 'cometd-nodejs-client'. '/node_modules/cometd-nodejs-client/cometd-nodejs-client.js' implicitly has an 'any' type.
  Try `npm install @types/cometd-nodejs-client` if it exists or add a new declaration (.d.ts) file containing `declare module 'cometd-nodejs-client';`

4 import { adapt } from 'cometd-nodejs-client';
                        ~~~~~~~~~~~~~~~~~~~~~~


Found 1 error.
```

In my opinion, the ideal solution to this problem would be for this library to provide a "TypeScript Declaration File" (`.d.ts`), in which exported functions and types are explicitly declared for use within TypeScript.

Is this something the project considers to be desirable? If so, please see #25 for a proposed solution.

--- a/.gitignore
+++ b/.gitignore
@@ -5,3 +5,4 @@
 .idea/
 
 node_modules/
+package-lock.json

--- a/README.md
+++ b/README.md
@@ -26,7 +26,7 @@ Therefore, you need to install the CometD JavaScript Client, version 3.1.2 or gr
 npm install cometd
 ```
 
-### Usage
+### Usage (CommonJS)
 
 ```javascript
 // Run the adapter code that implements XMLHttpRequest.
@@ -36,6 +36,18 @@ require('cometd-nodejs-client').adapt();
 var lib = require('cometd');
 var cometd = new lib.CometD();
 ...
+```
+
+### Usage (ES Modules)
+
+```javascript
+import { CometD } from 'cometd';
+import { adapt } from 'cometd-nodejs-client';
+// Shim XMLHTTPRequest for Node.js (required by CometD).
+adapt();
+
+// Your normal CometD client application here.
+const client = new CometD();
 ```
 
 See [here](https://github.com/cometd/cometd-javascript/blob/master/README.md) for an example CometD client application.

--- a/cometd-nodejs-client.d.ts
+++ b/cometd-nodejs-client.d.ts
@@ -1,0 +1,12 @@
+export interface Proxy {
+  uri?: string;
+  includes?: string[];
+  excludes?: string[];
+}
+
+export interface Options {
+  logLevel?: "debug" | "info";
+  httpProxy: Proxy;
+}
+
+export function adapt(options?: Options): void;

--- a/cometd-nodejs-client.d.ts
+++ b/cometd-nodejs-client.d.ts
@@ -1,4 +1,19 @@
-export interface Proxy {
+/*
+ * Copyright (c) 2017-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export interface HttpProxy {
   uri?: string;
   includes?: string[];
   excludes?: string[];
@@ -6,7 +21,7 @@ export interface Proxy {
 
 export interface Options {
   logLevel?: "debug" | "info";
-  httpProxy?: Proxy;
+  httpProxy?: HttpProxy;
 }
 
 export function adapt(options?: Options): void;

--- a/cometd-nodejs-client.d.ts
+++ b/cometd-nodejs-client.d.ts
@@ -6,7 +6,7 @@ export interface Proxy {
 
 export interface Options {
   logLevel?: "debug" | "info";
-  httpProxy: Proxy;
+  httpProxy?: Proxy;
 }
 
 export function adapt(options?: Options): void;

--- a/package.json
+++ b/package.json
@@ -13,6 +13,7 @@
   },
   "license": "Apache-2.0",
   "main": "cometd-nodejs-client.js",
+  "types": "cometd-nodejs-client.d.ts",
   "repository": {
     "type": "git",
     "url": "https://github.com/cometd/cometd-nodejs-client.git"

